extension/test/data_test.ts

```diff
@@ -63,4 +64,145 @@ describe('data.ts', function () {
       ).to.not.include.something.like({ entry: /^凡打 .*/ });
     });
   });
+  describe('kanjiSearch', function () {
+    it('should return null for kanji with char code < 0x3000', function () {
+      const result = rcxDict.kanjiSearch('A');
+      expect(result).to.be.null;
+    });
+
+    it('should return null if kanjiData entry is not properly formatted', function () {
+      const result = rcxDict.kanjiSearch('子9');
+      expect(result).to.be.null;
+    });
+
+    it('should return a DictEntryData object for valid kanji', function () {
+      const result: DictEntryData | null = rcxDict.kanjiSearch('日');
+      expect(result).to.deep.include({
+        kanji: '日',
+        eigo: 'day, sun, Japan, counter for days',
+      });
+    });
+
+    it('should set kanji property of DictEntryData object', function () {
+      const result = rcxDict.kanjiSearch('日');
+      expect(result?.kanji).to.equal('日');
+    });
+
+    it('should set misc property of DictEntryData object', function () {
+      const result = rcxDict.kanjiSearch('日');
+      expect(result?.misc).to.deep.include({
```

Comment:
Style preference:
Try to separate tests into 2 or 3 blocks:

```
Arrange

Act

Assert
```

With a newline between each one (somtimes arrange won't exist or act and assert will be grouped).

Used consistently it makes it very easy to understand tests we read for the first time.